running/architecture.rst

```diff
@@ -2,3 +2,72 @@ Architecture
 ============
 
 Deconst is distributed as a set of Docker containers, deployed by an Ansible playbook.
+
+Terminology
+-----------
+
+.. glossary::
+
+  content repository
+    Version controlled repository containing markup that provides content for some subset of the
+    completed site. To begin, we'll assume that all of these are in ``git`` repositories on
+    `GitHub <github.com>`_. Most of our documentation will be in Sphinx reStructuredText, but the
+    developer blog will be in Markdown as part of a Jekyll site.
+
+  map repository
+    Version controlled repository containing plain-text documents that associate subtrees of
+    indexed content, identified by a :term:`content ID` prefix, with subtrees of :term:`paths`
+    on the presented site.
+
+  content ID
+    Unique identifier assigned to a single page of content. By convention, these are URLs that
+    join the https URL of the GitHub repository, ``tree/branch`` if "branch" is not master, and
+    the subpath of the page within the *rendered* output of the documentation build tool. Most of
+    the architecture should treat these as opaque strings. Examples:
+    this page would be ``https://github.com/deconst/deconst-docs/running/architecture`` on master,
+    or ``https://github.com/deconst/deconst-docs/tree/glossary/running/architecture`` until this
+    pull request is merged.
```

Comment:
Oh, yeah, whoops, I definitely made that too specific. I meant the GitHub-URL bits here to be an illustration with a concrete example.

One consideration: it's not the URL of the source document - it's the base URL of the content repository plus the path to a page within the *rendered* output, which doesn't necessarily correspond to a real file in the repository. You can build a Sphinx repository into a single page, for example, in which case the whole thing would have a single content ID. Blog posts in the Jekyll site will also have content IDs that differ from the source document's URL because they live in `_posts` but are rendered elsewhere.